apps/src/lib/ui/lti/sync/LtiSectionSyncDialog.tsx

```diff
@@ -1,24 +1,30 @@
-import PropTypes from 'prop-types';
-import React, {useState} from 'react';
+import React, {CSSProperties, useState} from 'react';
 import i18n from '@cdo/locale';
 import BaseDialog from '@cdo/apps/templates/BaseDialog';
 import Button from '@cdo/apps/templates/Button';
 import DialogFooter from '@cdo/apps/templates/teacherDashboard/DialogFooter';
 import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
 import Spinner from '@cdo/apps/code-studio/pd/components/spinner';
+import {
+  LtiSectionSyncDialogProps,
+  LtiSectionSyncResult,
+  SubView,
+} from './types';
+import PropTypes from 'prop-types';
+import $ from 'jquery';
 
 // This dialog is shown to the teacher whenever they have requested Code.org to
 // import/sync the teacher's sections and students managed by their LMS.
-export default function LtiSectionSyncDialog(props) {
-  const SYNC_RESULT_VIEW = 'syncResult';
-  const SPINNER_VIEW = 'spinner';
-  const ERROR_VIEW = 'error';
-  const {syncResult, onClose} = props;
-  const initialView = syncResult.error ? ERROR_VIEW : SYNC_RESULT_VIEW;
-  const [currentView, setCurrentView] = useState(initialView);
+export default function LtiSectionSyncDialog({
+  syncResult,
+  onClose,
+  disableRosterSyncEnabled,
```

Comment:
The name `disableRosterSyncEnabled` is a bit confusing to me. Can you explain what this prop does?